packages/server/src/script/resendSignupEmails.ts

```diff
@@ -0,0 +1,136 @@
+
+        const createdFriendlyString = info.created.toDate().toISOString().split('T')[0]
+        const resendReasonAndOriginalSignupDate = resendReason
+            .concat(` Voter originally signed up for VoteByMail.io on ${createdFriendlyString}`)
+        const letter = new Letter(info, method, info.id, resendReasonAndOriginalSignupDate)
+        const pdfBuffer = await toPdfBuffer(letter.render('html'), await letter.form)
+
+        // Send emails.
+        const mgResponse = await sendSignupEmail(
+            letter,
+            info.email,
+            method.emails,
+            { pdfBuffer },
+        )
+        if (!mgResponse || mgResponse.message != 'Queued. Thank you.') {
+            console.log('Error sending email for ' + info.id + '. Exiting.')
+            console.log(mgResponse)
+            return
+        }
+        console.log('Successfully sent emails for ' + info.id)
+
+
+        // Number of resends.  _0 is the first, _1 the second, etc.
+        const numResends = info.resends ? info.resends.length : 0
+        const file = storageFileFromId(info.id.concat('-') + numResends)
+        await file.upload(pdfBuffer)
+
+        // Send faxes
+        let twilioResponses: TwilioResponse[] = []
+        if (method.faxes.length > 0) {
+            const uri = await file.getSignedUrl(24 * 60 * 60 * 1000)
+            const resposnes = await sendFaxes(uri, method.faxes)
+            twilioResponses = resposnes.map(({url, sid, status}) => ({url, sid, status}))
+
+            for (const twilioResponse of twilioResponses) {
+                if (twilioResponse.status != "queued") {
+                    console.log('Error sending faxes for ' + info.id + '. Exiting.')
+                    console.log(twilioResponse)
+                    return
+                }
+            }
+          }
+
+        // Update voter data in firestore.
+        const resend = { dateTime: new Date().getTime(), reason: resendReason }
+        await firestoreService
+            .db
+            .collection('StateInfo')
+            .doc(info.id)
+            .update({
+                resends: admin.firestore.FieldValue.arrayUnion(resend)
+            })
+    }
+}
+
+// Reads './data/conditions.ts' and resends emails to all voter sign ups that satisfy query.
+const main = async () => {
+    // Fetch registrations.
+    const query: Query = newYorkQuery
+    const registrations = await firestoreService.query<RichStateInfo>(query)
+    if (registrations == null || registrations.length == 0) {
+        console.log('No voters found for specified conditions.')
+        return
+    }
+
+    // Display registration previews.
+    const registrationsToDisplay = registrations.length > 10 ? registrations.slice(0, 10) : registrations
+    console.log('There are a total of %s entries for your query.  Here is a sample:', registrations.length)
+    for (const registration of registrationsToDisplay) {
+        const previewMap = new Map(Object.entries(registration))
```

Comment:
I finally figured out what this function does.  The typesafe (and clearer) way to do this is just to write a function
```ts
const selectPreviewFields(richState: RichStateInfo) => Record<string, string>
```
and perform all your logic in there.  Then you can make it very clear what field's you are selecting and put this into `conditions.sample.ts` as a function.

Reply:
Followin up here.